scripts/test-w-coverage.sh

```diff
@@ -2,15 +2,24 @@
 
 set -aueo pipefail
 
-readarray -t modules < <(go list ./... | grep -v tests/framework | grep -v tests/e2e | grep -v tests/scenarios | grep -v tests/scale | \
-   grep -v ci/ | grep -v demo/ | grep -v experimental/ | grep -v scripts/)
+readarray -t modules < <(go list ./... | \
+                             grep -v tests/framework | \
+                             grep -v tests/e2e | \
+                             grep -v tests/scenarios | \
+                             grep -v tests/scale | \
+                             grep -v ci/ | \
+                             grep -v demo/ | \
+                             grep -v experimental/ | \
+                             grep -v scripts/)
 
 go test -timeout 120s \
    -failfast \
    -v \
-   -coverprofile=coverage.txt \
+   -coverprofile=coverage.txt.with_generated_code \
    -covermode count "${modules[@]}" | tee testoutput.txt || { echo "go test returned non-zero"; exit 1; }
 
+cat coverage.txt.with_generated_code | grep -v "_generated.go" > coverage.txt
```

Comment:
Instead of suffixing `_generated` we could have matched on `mock_` pattern and skipped file renames, wondering  if you preferred one over the other and why :-)

Reply:
@shashankram I thought about that and decided that you could have `mock_` files that are not generated. On the other hand `_generated.go` is clearly something we should not test for coverage.